### PR TITLE
Add a dockerfile that builds from source.

### PR DESCRIPTION
**Pull request by mstemm**
Add a Dockerfile that builds from a checked out tree, loosely
following the steps implied by
https://github.com/nodejs/docker-node/blob/master/4.3/onbuild/Dockerfile:

 - copy package.json to image
 - run npm install within image
 - copy source code to image

Continue to use /opt/juttle-engine/juttles as the root for consistency with
the released image.

This npm install does not install globally, so symlink
juttle/juttle-engine to /usr/local/bin so they can be run without a
path.

This PR also bumps the juttle-cloudwatch-adapter to version 0.2.0. I
can pull that out into a separate PR if necessary.

I've also set up automated builds from docker hub for the juttle-engine repo. There's even one there right now for this branch! https://hub.docker.com/r/juttle/juttle-engine-canary/tags/

By default the branch name will be used as the tag.

@rlgomes @mnibecker

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
+FROM node:4.2.4-wheezy
+MAINTAINER Mark Stemm (mstemm@jut.io)
+
+RUN mkdir -p /opt/juttle-engine
+WORKDIR /opt/juttle-engine
+ADD package.json package.json
+RUN npm install
+COPY . .
```
Comment (demmer) [inline Dockerfile]: I'm no docker expert but it seems to me you should do this before running npm install, then you don't need the `ADD package.json package.json` line above, right?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+RUN mkdir -p /opt/juttle-engine/juttles
+COPY examples /opt/juttle-engine/juttles/examples
+
+# Also symlink juttle and juttle-engine so they can be run directly
+# without specifying a path.
+RUN ln -s /opt/juttle-engine/bin/juttle /usr/local/bin/juttle
```
Comment (demmer) [inline Dockerfile]: You should be able to run `npm link` to accomplish this same thing but without having to explicitly enumerate the scripts.

File: package.json [hunk 3]
```diff
@@ -35,7 +35,7 @@
     "express": "^4.13.4",
     "juttle": "^0.4.2",
     "juttle-aws-adapter": "^0.1.3",
-    "juttle-cloudwatch-adapter": "^0.1.2",
+    "juttle-cloudwatch-adapter": "^0.2.0",
```
Comment (demmer) [inline package.json]: This seems like an unrelated change.

File: Dockerfile (hunk 1 above)
Comment (mstemm) [inline Dockerfile]: The wisdom of the crowds say this order allows for better caching, as it's more likely that the results of npm install is stable than the code you're changing.

File: package.json (hunk 3 above)
Comment (mstemm) [inline package.json]: yes "This PR also bumps the juttle-cloudwatch-adapter to version 0.2.0. I
can pull that out into a separate PR if necessary."

File: Dockerfile (hunk 2 above)
Comment (mstemm) [inline Dockerfile]: ok will try that.

**Comment by mstemm** [comment]
Ok moved changes other than the dockerfile itself to a separate PR. @demmer can you look again?

**Comment by demmer** [comment]
Lgtm
